source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -110,6 +113,40 @@ namespace aspect
       return phi_theta_d;
     }
 
+    template <int dim>
+    Point<3>
+    EllipsoidalChunk<dim>::EllipsoidalChunkGeometry::push_forward_topography(const Point<3> &phi_theta_d_hat) const
+    {
+      const double d_hat = phi_theta_d_hat[2]; // long, lat, depth
+      Point<dim-1> phi_theta;
+      const double rad_to_degree = 180/numbers::PI;
+      if (dim == 3)
+        phi_theta = Point<dim-1>(phi_theta_d_hat[0] * rad_to_degree,phi_theta_d_hat[1] * rad_to_degree);
+      const double h     = topography->value(phi_theta);
```

Comment:
The problem here is that this function is called by the push_forward(const Point<3> &chart_point) function, for which I would also need to add this, and because push_forward is defined by the ChartManifold class, I can't just change the arguments. Although I could try a bind, but I am not sure how to do that properly in this case. I could also try overloading the push_forward function, but I think this also isn't a clean option because they could be used elsewhere.

Furthermore, this function might be used after the first timestep for, for example, the representative point or post processors.

Could you explain in more detail how you would see this?